charts/argo-workflows/templates/controller/workflow-rb.yaml

```diff
@@ -1,19 +1,20 @@
 {{- if .Values.workflow.rbac.create -}}
+  {{- range $namespace := .Values.singeNamespace | ternary (list "") (append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq) }}
+---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: {{ template "argo-workflows.fullname" . }}-workflow
-  {{- with .Values.workflow.namespace }}
-  namespace: {{ . }} 
-  {{- end }}
+  name: {{ template "argo-workflows.fullname" $ }}-workflow
+  namespace: {{ $namespace }}
```

Comment:
Please wrap namespace in `{{with}}` here, same as for roles.

Reply:
Fixed in https://github.com/argoproj/argo-helm/pull/824/commits/2bfcf1bc0e9a031dbe5c5ff0c0e9d524da742f75